backend/api/Services/EmergencyActionService.cs

```diff
@@ -0,0 +1,42 @@
+﻿using Api.Services.Events;
+namespace Api.Services
+{
+    public interface IEmergencyActionService
+    {
+        public void TriggerEmergencyButtonPressedForRobot(EmergencyButtonPressedForRobotEventArgs e);
+
+        public void TriggerEmergencyButtonDepressedForRobot(EmergencyButtonPressedForRobotEventArgs e);
+    }
+
+    public class EmergencyActionService : IEmergencyActionService
+    {
+
+        public EmergencyActionService()
+        {
+        }
+
+        public void TriggerEmergencyButtonPressedForRobot(EmergencyButtonPressedForRobotEventArgs e)
+        {
+            OnEmergencyButtonPressedForRobot(e);
+        }
+
+        public void TriggerEmergencyButtonDepressedForRobot(EmergencyButtonPressedForRobotEventArgs e)
+        {
+            OnEmergencyButtonDepressedForRobot(e);
+        }
+
+        public static event EventHandler<EmergencyButtonPressedForRobotEventArgs>? EmergencyButtonPressedForRobot;
+
+        protected virtual void OnEmergencyButtonPressedForRobot(EmergencyButtonPressedForRobotEventArgs e)
+        {
+            EmergencyButtonPressedForRobot?.Invoke(this, e);
```

Comment:
Does the ? do more than suppress the type warning for the possible null reference? Is this guaranteed or should we do a check to verify it?

Reply:
@aestene did the work on events, i'm not entirely sure about this